LanguageFeatures/Digit_separators/digit_separators_A04_t06.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Allow one or more `_`s between any two otherwise adjacent digits
+/// of a `NUMBER` or `HEX_NUMBER` token. The following are not digits: The
+/// leading `0x` or `0X` in `HEX_NUMBER`, and any `.`, `e`, `E`, `+` or `-` in
+/// `NUMBER`.
+///
+/// That means only allowing `_`s between two `0-9` digits in `NUMBER` and
+/// between two `0-9`, `a-f`, `A-F` digits in `HEX_NUMBER`.
+///
+/// @description Check that it is a compile-time error if `_` follows or is
+/// followed by `+` or `-`.
+/// @author sgrekhov22@gmail.com
+
+const pi1 = 0.0314e_+2;
+//          ^^^^^^^^^^
```

Comment:
Is this error marking actually precise? I would expect `0.0314e_` to be an unrecognized token (a lexer error, no less), or a syntax error (a literalInteger followed by an identifier `e_`), and `+2` would in any case be a perfectly legit continuation of the expression.

Reply:
Yes, it is presize. Both analyzer and VM produce here `Digit separators ('_') in a number literal can only be placed between two digits. Try removing the '_'.`  `0.0314` here cannot be a `literalInteger`, it contains `.`. But on the line 22 below `314` can be. But anyway both of the tools produce the same error on line 22 as well. Added
```dart
const e_ = 0;
const E_ = 0;
```
to make the test stronger.

Reply:
Very good, thanks!